src/main/kotlin/processing/filters/Blend.kt

```diff
@@ -0,0 +1,135 @@
+@file:UseSerializers(ImageSerializer::class)
+
+package processing.filters
+
+import javafx.embed.swing.SwingFXUtils
+import javafx.scene.image.Image
+import javafx.scene.image.PixelReader
+import javafx.scene.image.PixelWriter
+import javafx.scene.image.WritableImage
+import javafx.scene.paint.Color
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+import kotlinx.serialization.builtins.ByteArraySerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import processing.ImageProcessing
+import java.io.ByteArrayOutputStream
+import javax.imageio.ImageIO
+import kotlin.math.absoluteValue
+import kotlin.math.sqrt
+import kotlin.random.Random
+
+@Serializable
+@SerialName("Blend")
+class Blend(private val blendImage: Image, val mode: BlendType) :
+    ImageProcessing {
+    override fun process(image: WritableImage) {
+        val width = image.width.toInt().coerceAtMost(blendImage.width.toInt())
+        val height = image.height.toInt().coerceAtMost(blendImage.height.toInt())
+        val readerA: PixelReader = blendImage.pixelReader
+        val readerB: PixelReader = image.pixelReader
+        val writer: PixelWriter = image.pixelWriter
+        for (y in 0 until height) {
+            for (x in 0 until width) {
+                val oldColorA = readerA.getColor(x, y)
+                val oldColorB = readerB.getColor(x, y)
+                writer.setColor(x, y, mode.operation(oldColorA, oldColorB))
+            }
+        }
+    }
+}
+
+enum class BlendType(val operation: (Color, Color) -> Color) {
```

Comment:
Will it be better if you have some comment of the params, especially the two `Color` parameters in the function given. E.g.

```kotlin
/* @param operation: (blendImagePixel, baseImagePixel) -> resultImagePixel */
```